### ISSUE TITLE
[FR] Suggestions for optimizing the display of PDF annotations

### ISSUE THREAD

**Issue by dfyxandskwy** [RyotaUshio/obsidian-pdf-plus#89]
### Scenario Description
In PDF reading, users tend to highlight and underline noteworthy content within the page. If the number of these contents in a PDF page is not much, and a knowledge of the text is not too long, the entire PDF page does not look too fancy, then, the page highlighting can indeed effectively attract the attention of users. However, once the content is too much, too long, highlighting will lose its effect. In order to reduce the page highlighting, users will be in the annotation, only highlight the keyword words, rather than the entire sentence.
Our plugin in order to achieve the transfer of text in the PDF to the obsidian document, the current annotation method is to highlight the entire sentence to get a deep link. In this way, it will encounter what I said above highlighting too much content, too long.

### **Idea**
Users can get the deep link of the corresponding text without highlighting the PDF text and place it in the obsidian document. Then, the user can highlight and underline some keywords in the PDF text just obtained in the obsidian document. Then with the help of our plugin, the highlighting and underlining of these partial keywords can be reversely displayed on the PDF page.

### Advantages
If the above idea is realized, the PDF page display will be more concise, and the user can only through an annotation operation, in the obsidian document and PDF page in two places to achieve the keyword highlighting, underlining.

### Existing conditions
I think the realization of the above ideas need to use the obsidian PDF reader in the deep links, backlinks feature. Fortunately, PDF + + currently take full advantage of these two features. PDF + + should be more likely than obsidian in other PDF plugins to realize my ideas.

At present, I have not encountered a PDF reader or PDF plugin that can realize the above idea. I hope you will consider my suggestions. Good luck with your work.

**Comment by RyotaUshio** [comment]
Thanks a lot for the detailed suggestion, and I totally get your point.
I guess your idea can be already realizable with the following approach:


First, create & enable the following CSS snippet.

```css
.pdf-plus-backlink-selection {
    background-color: transparent !important;
}

.pdf-plus-backlink-selection:hover {
    background-color: green !important;
}
```

Then you will be able to do something like this:

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/5b7b2b7f-c981-4877-8687-089c91d3bf36

**Comment by RyotaUshio** [comment]
Update: From next version on, the following snippet will give better results for multi-line highlights:

```css
.pdf-plus-backlink-selection:not(.is-hovered) {
    background-color: transparent !important;
}

.pdf-plus-backlink-selection.is-hovered {
    opacity: 0.1 !important;
}
```

**Comment by RyotaUshio** [comment]
The `is-hovered` CSS class (see my previous comment) is now available in [0.37.12](https://github.com/RyotaUshio/obsidian-pdf-plus/releases/tag/0.37.12).
